include/deal.II/differentiation/ad/ad_drivers.h

```diff
@@ -815,6 +1116,74 @@ namespace Differentiation
                    independent_variables.data(),
                    J.data());
       }
+
+    protected:
+      /**
+       * Index of the tape that is currently in use. It is this tape that will
+       * be recorded to or read from when performing computations using "taped"
+       * auto-differentiable numbers.
+       */
+      types::tape_index active_tape;
+
+      /**
+       * Mark whether we're going to inform taped data structures to retain
+       * the coefficients ("Taylors" in ADOL-C nomenclature) stored on the
+       * tape so that they can be evaluated again at a later stage.
+       */
+      bool keep_values;
+
+      /**
+       * Mark whether we're currently recording a tape. Dependent on the state
+       * of this flag, only a restricted set of operations are allowable.
+       */
+      bool is_recording_flag;
+
+      /**
+       * A flag indicating that we should preferentially use the user-defined
+       * taped buffer sizes as opposed to either the default values selected
+       * by the AD library (or, in the case of ADOL-C, defined in an
+       * ".adolcrc" file).
+       */
+      bool use_stored_taped_buffer_sizes;
+
+      /**
+       * ADOL-C operations buffer size.
+       */
+      types::tape_buffer_sizes obufsize;
+
+      /**
+       * ADOL-C locations buffer size.
+       */
+      types::tape_buffer_sizes lbufsize;
+
+      /**
+       * ADOL-C value buffer size.
+       */
+      types::tape_buffer_sizes vbufsize;
+
+      /**
+       * ADOL-C Taylor buffer size.
+       */
+      types::tape_buffer_sizes tbufsize;
+
+      /**
+       * Return a list of registered tape indices
+       */
+      std::vector<types::tape_index>
+      get_registered_tape_indices() const
+      {
+        // It is unknown as to why this should not be the same type
+        // as types::tape_index...
+        std::vector<short> registered_tape_indices_s;
+        cachedTraceTags(registered_tape_indices_s);
```

Comment:
Can you explain the comment? What does ADOL-C use as the argument data type for `cachedTraceTags`? And is it a bug in their own interfaces that it is not equal to `types::tape_index`, or do they uniformly use `short`?

Reply:
Oh, right. My comment is incorrect. They do use [`short`](https://gitlab.com/adol-c/adol-c/blob/master/ADOL-C/include/adolc/taping.h#L100) for tape numbering, but the valid range for a tape is `0 < tape_index <`[`TBUFNUM`](https://gitlab.com/adol-c/adol-c/blob/master/ADOL-C/include/adolc/internal/usrparms.h#L34). So I chose to use `unsigned short` to prevent negative values for inputs. I'll remove this comment (although I prefer to keep the use of an unsigned type, if you don't disagree too strongly).

Reply:
The code is fine by me. I would never have said anything if there was not a comment I didn't understand :-)